control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -3480,3 +3491,237 @@ func healthCheckIdentityProvider(ctx context.Context, hcp *hyperv1.HostedControl
+}
+
+func createAWSDefaultSecurityGroup(ctx context.Context, ec2Client ec2iface.EC2API, infraID, vpcID string) (string, error) {
+	logger := ctrl.LoggerFrom(ctx)
+
+	// Determine VPC cidr
+	vpcResult, err := ec2Client.DescribeVpcsWithContext(ctx, &ec2.DescribeVpcsInput{
+		VpcIds: []*string{awssdk.String(vpcID)},
+	})
+	if err != nil {
+		logger.Error(err, "Failed to describe vpc", "vpcID", vpcID)
+		return "", fmt.Errorf("failed to describe vpc %s, code %s", vpcID, awsErrorCode(err))
+	}
+	if len(vpcResult.Vpcs) == 0 {
+		return "", fmt.Errorf("vpc %s not found", vpcID)
+	}
+	vpcCIDR := awssdk.StringValue(vpcResult.Vpcs[0].CidrBlock)
+	describeSGResult, err := ec2Client.DescribeSecurityGroups(&ec2.DescribeSecurityGroupsInput{Filters: awsSecurityGroupFilters(infraID)})
+	if err != nil {
+		logger.Error(err, "Failed to list security groups")
+		return "", fmt.Errorf("cannot list security groups, code: %s", awsErrorCode(err))
+	}
+	sgID := ""
+	var sg *ec2.SecurityGroup
+	if len(describeSGResult.SecurityGroups) > 0 {
+		sg = describeSGResult.SecurityGroups[0]
+		sgID = awssdk.StringValue(sg.GroupId)
+	}
+	if sgID == "" {
+		// Create a security group if one is not found
+		tags := []*ec2.Tag{
+			{
+				Key:   awssdk.String(fmt.Sprintf("kubernetes.io/cluster/%s", infraID)),
+				Value: awssdk.String("owned"),
+			},
+			{
+				Key:   awssdk.String("Name"),
+				Value: awssdk.String(awsSecurityGroupName(infraID)),
+			},
+		}
+		createSGResult, err := ec2Client.CreateSecurityGroup(&ec2.CreateSecurityGroupInput{
+			GroupName:   awssdk.String(awsSecurityGroupName(infraID)),
+			Description: awssdk.String("default worker security group"),
+			VpcId:       awssdk.String(vpcID),
+			TagSpecifications: []*ec2.TagSpecification{
+				{
+					ResourceType: awssdk.String("security-group"),
+					Tags:         tags,
+				},
+			},
+		})
+		if err != nil {
+			logger.Error(err, "Failed to create security group")
+			return "", fmt.Errorf("failed to create security group, code: %s", awsErrorCode(err))
+		}
+		sgID = awssdk.StringValue(createSGResult.GroupId)
+
+		// Fetch just-created SG
+		backoff := wait.Backoff{
+			Steps:    5,
+			Duration: 1 * time.Second,
+			Factor:   1.0,
+			Jitter:   0.1,
+		}
+		err = retry.OnError(backoff, func(error) bool { return true }, func() error {
+			var err error
+			describeSGResult, err = ec2Client.DescribeSecurityGroups(&ec2.DescribeSecurityGroupsInput{
+				GroupIds: []*string{awssdk.String(sgID)},
+			})
+			if err != nil || len(describeSGResult.SecurityGroups) == 0 {
+				return fmt.Errorf("cannot find SG yet")
+			}
+			return nil
+		})
+		if err != nil {
+			return "", fmt.Errorf("cannot find security group that was just created %s", sgID)
+		}
+		sg = describeSGResult.SecurityGroups[0]
+		logger.Info("Created security group", "id", sgID)
+	}
+	egressPermissions := supportawsutil.DefaultWorkerSGEgressRules()
+	_, err = ec2Client.AuthorizeSecurityGroupEgress(&ec2.AuthorizeSecurityGroupEgressInput{
```

Comment:
My experience is that the result of a `CreateSecurityGroup` is a security group with this rule already:

```go
func DefaultWorkerSGEgressRules() []*ec2.IpPermission {
	return []*ec2.IpPermission{
		{
			IpProtocol: aws.String("-1"),
			IpRanges: []*ec2.IpRange{
				{
					CidrIp: aws.String("0.0.0.0/0"),
				},
			},
		},
	}
}
```
Does this always result in an error?

Reply:
At least we typically have to always revoke this rule when not desired, e.g. https://github.com/openshift/osd-network-verifier/blob/86efe5d4069fcc5efecae5e744a1a42e0233b4f5/pkg/verifier/aws/aws_verifier.go#L433-L451

Reply:
Good to know. I will test leaving it out and if that works fine, then will remove it.